### ISSUE TITLE
Не могу авторизоваться с учетными данными созданного пользователя

### ISSUE THREAD

**Issue by krkub3** [greatbit/quack#203]
**Исходные данные**
Развернут QuAck на standalone сервере.

**Создание пользователя**
1. Аутентификация с учетными данными root/rootpass
2. В правом верхнем углу нажимаю root --> Create User
3. Заполняю поле Login : testuser
4. Заполняю поле Password : testuser
5. Нажимаю Create
6. Выхожу (Log out)

При авторизации с учетными данными testuser/testuser получаю ошибку:
_Unable to login: { "message" : "ru.greatbit.whoru.auth.error.UnauthorizedException: Login and password should match", "code" : 401 }_

**Comment by azee** [comment]
Проверьте вот эту проперти
https://github.com/greatbit/quack/blob/master/assembly/quack.properties#L8

По дефолту авторизация тупая - для демо и только через проперти.
Если установить в
com.testquack.api.security.DbAuthProvider
то будет работать создание пользователей в базе и можно будет авторизовываться новыми пользователями

Пока у вас один инстанс, можно использовать InmemSessionProvider
https://github.com/greatbit/quack/blob/master/assembly/quack.properties#L12

Но если у вас будет много реплик, то нужно будет поменять конфиг на 
ru.greatbit.whoru.auth.providers.HazelcastSessionProvider
и установить конфигурацию хазелкаста, перечислив ноды
https://github.com/greatbit/quack/blob/master/assembly/quack.properties#L29
Можно с wildcards (если у вас настроен VPC) 
Например
hazelcast.interface=10.0.*.*
Это нужно, чтобы сессии пользователей хранились не в памяти на каждой машине, а в распределённом хранилище. Хазелкаст - inmemory database - быстрая и позволяет не нагружать основную БД на каждый запрос к сессии

**Comment by azee** [comment]
AuthProcvider интерфейс сделан для того, чтобы каждая компания могла авторизацию под себя написать, если внутри используется что-то кастомное. 
Есть готовый провайдер для AWS Cognito. Был рабочий вариант для JiraServer. Сейчас работаем над SSO.

**Comment by krkub3** [comment]
Если я устанавливаю _whoru.session.provider=com.testquack.api.security.DbAuthProvider_, то авторизация новых пользователей работает! Благодарю!

Сразу вопрос тогда, какие параметры из конфигурационного файла обязательны, а какие можно убрать?
Например, нужны ли эти параметры:
```
auth.admin.token=abc
auth.admin.logins=azee,zyxel560k@mail.ru
storage.provider=com.testquack.storage.MongoStorage
quack.tracker=com.testquack.tracker.DummyTracker
```

**Comment by krkub3** [comment]
И ещё один вопрос. Как удалить созданного пользователя?

**Comment by azee** [comment]
Да. При DBAuthProvider проекты создают админы и добавляют пользователей в списки доступа в проекты.

Удалить пользователя сейчас можно либо через базу, либо через api call от имени админа (надо передать token, установленный в конфиге)

Если есть такая потребность - можем сделать UI для удаления пользователей.

Если же есть желание дать возможность пользователям создавать проекты - то можно включить функциональность Организаций. Пользователь может создать организацию и становится её админом. Внутри организации он может создавать проекты. На это всё можно посмотреть в облаке http://cloud.testquack.com/ (3 параллельные пользовательские сессии на организацию - бесплатно)

**Comment by krkub3** [comment]
_Если есть такая потребность - можем сделать UI для удаления пользователей._
Это было бы восхитительно!

**Comment by krkub3** [comment]
> Проверьте вот эту проперти https://github.com/greatbit/quack/blob/master/assembly/quack.properties#L8
> 
> По дефолту авторизация тупая - для демо и только через проперти. Если установить в com.testquack.api.security.DbAuthProvider то будет работать создание пользователей в базе и можно будет авторизовываться новыми пользователями
> 
> Пока у вас один инстанс, можно использовать InmemSessionProvider https://github.com/greatbit/quack/blob/master/assembly/quack.properties#L12
> 
> Но если у вас будет много реплик, то нужно будет поменять конфиг на ru.greatbit.whoru.auth.providers.HazelcastSessionProvider и установить конфигурацию хазелкаста, перечислив ноды https://github.com/greatbit/quack/blob/master/assembly/quack.properties#L29 Можно с wildcards (если у вас настроен VPC) Например hazelcast.interface=10.0._._ Это нужно, чтобы сессии пользователей хранились не в памяти на каждой машине, а в распределённом хранилище. Хазелкаст - inmemory database - быстрая и позволяет не нагружать основную БД на каждый запрос к сессии

Если я в конфигурационном файле напишу _whoru.auth.provider=ru.greatbit.whoru.auth.providers.StubAuthProvider_. Можно ли в конфигурационном файле создать несколько stub.login?
Например:
```
stub.login1=somelogin1
stub.password1=somepass1
stub.login2=somelogin2
stub.password2=somepass2
```
